### PR TITLE
733_Flood_Fill

### LINKED ISSUE #64: 733_Flood_Fill
## Summary
733_Flood_Fill
## Goal
733_Flood_Fill
## Todo

- [ ] add 733_Flood_Fill
- [ ] write time and space complexity

## Deadline
2/8
## Parent issue
https://github.com/okuda-seminar/Engineering/issues/599
## References
None

## Notes
None

### PR DESCRIPTION
**Pull request by Rwatana**
Please check my code.

### CODE REVIEW

**Comment by ryujumoto** [comment]
Why did you use Python 2 instead of Python 3?
Also, follow the coding convention PEP8, and follow Google style when writing a doc string.
[This code](https://github.com/okuda-seminar/leetcode_answers/blob/main/733_flood_fill.py) is an example.

File: 733_Flood_Fill.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+class Solution:
+
+    def floodFill(
+        self,
+        image: List[List[int]],
+        sr: int,
+        sc: int,
+        color: int
+            ) -> List[List[int]]:
```
Comment (ryujumoto) [inline 733_Flood_Fill.py:17]: Fix the indent.

File: 733_Flood_Fill.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        queue.append((sr, sc))
+        visited = set()
+        oldColor = image[sr][sc]
+
+        while queue:
+            sr, sc = queue.popleft()
+            image[sr][sc] = color
+            for direction in directions:
+                newSr, newSc = sr + direction[0], sc + direction[1]
+                if (newSr >= 0) and (newSr < len(image)) and (newSc >= 0) and (newSc < len(image[0])) and ((newSr, newSc) not in visited) and (image[newSr][newSc] == oldColor):
```
Comment (ryujumoto) [inline 733_Flood_Fill.py]: Follow the PEP8, namely, variable names should be snake case, and the length of one line should be less than 81.

**Comment by Rwatana** [comment]
I fixed the code. Please check.

In vscode, on line 45-48 I got an error.
But, I follow the [pep8](https://pep8-ja.readthedocs.io/ja/latest/)
Please tell me whether I have to improve them or not.

**Comment by ryujumoto** [comment]
@Rwatana 
You can avoid the error by enclosing the conditions of the if statement in parentheses.

**Comment by Rwatana** [comment]
I think I could fix the error. Please check.

File: 733_Flood_Fill.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            color (int): Replacement color
+
+        Returns:
+            List[List[int]]: 'image' after replaced with `color`
+        """
+        directions = [(0, -1), (0, 1), (1, 0), (-1, 0)]
+        queue = deque()
+        queue.append((start_row, start_colum))
+        visited = set()
+        oldColor = image[start_row][start_colum]
```
Comment (ryujumoto) [inline 733_Flood_Fill.py:38]: oldColor -> old_color

File: 733_Flood_Fill.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            image[start_row][start_colum] = color
+            for direction in directions:
+                new_start_row = start_row + direction[0]
+                new_start_colum = start_colum + direction[1]
+                if ((new_start_row >= 0) and
+                    (new_start_row < len(image)) and
+                    (new_start_colum >= 0) and
+                    (new_start_colum < len(image[0])) and
+                    ((new_start_row, new_start_colum) not in visited) and
+                        (image[new_start_row][new_start_colum] == oldColor)):
```
Comment (Ryuju0911) [inline 733_Flood_Fill.py:51]: Fix the indent.

**Pull request closed without merging**